File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    for g in goal:
        if cards1 and g == cards1[0]:
            cards1.remove(g)
        elif cards2 and g == cards2[0]:
            cards2.remove(g)
        else:
            return "No"
    return "Yes"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "";
        int c1 = 0;
        int c2 = 0;
        
        for(String s : goal){
            if(c1 < cards1.length && cards1[c1].equals(s)){
                c1++;
                continue;
            }
            else if(c2 < cards2.length && cards2[c2].equals(s)){
                c2++;
                continue;
            } else{
                return "No";
            }
        }
        return "Yes";
    }
}